packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -1108,20 +1240,53 @@ class MediaQuery extends InheritedModel<_MediaQueryAspect> {
   /// the [MediaQueryData.devicePixelRatio] property of the ancestor [MediaQuery] changes.
   static double? maybeDevicePixelRatioOf(BuildContext context) => _maybeOf(context, _MediaQueryAspect.devicePixelRatio)?.devicePixelRatio;
 
+  /// Deprecated. Will be removed in a future version of Flutter. Use
+  /// [maybeTextScalerOf] instead.
+  ///
   /// Returns textScaleFactor for the nearest MediaQuery ancestor or
   /// 1.0, if no such ancestor exists.
   ///
   /// Use of this method will cause the given [context] to rebuild any time that
   /// the [MediaQueryData.textScaleFactor] property of the ancestor [MediaQuery] changes.
+  @Deprecated(
+    'Use textScalerOf instead. '
+    'Use of textScaleFactor was deprecated in preparation for the upcoming nonlinear text scaling support. '
+    'This feature was deprecated after v3.12.0-2.0.pre.',
+  )
   static double textScaleFactorOf(BuildContext context) => maybeTextScaleFactorOf(context) ?? 1.0;
 
+  /// Deprecated. Will be removed in a future version of Flutter. Use
+  /// [maybeTextScalerOf] instead.
+  ///
```

Comment:
Is this necessary given there is already a `@Deprecated` annoatation?

Reply:
I followed an example in media_query.dart but now I noticed not all deprecated properties have this. I guess it should be fine?

Reply:
That's my impression.  In my PRs I haven't changed the docs like this, just added `@Deprectated`.